--- a/python/CFDataFile.py
+++ b/python/CFDataFile.py
@@ -77,10 +77,8 @@ class CFDataFile:
         except:
             raise Exception('Unable to open %s for writing' % self.datafile_name)
         else:
             header = dic['#']
-            del dic['#']
-            index = list(dic.keys())
-            index.sort()
+            index = sorted(k for k in dic if k != '#')
             contents = '#|%s\n' %('|'.join(header))
             file.write(contents)
             for entry in index:
